### PR TITLE
endpoint: Fix label replacement

### PR DESCRIPTION
**Pull request by jrajahalme**
Fix bugs in label updates:
- Information labels were never deleted
- Information label assignments were logged even when nothing changed
- Identity label updates never updated the Source or Value of an existing label

Add unit test for label updates.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: pkg/endpoint/endpoint_test.go [hunk 1]
```diff
@@ -171,6 +172,44 @@ func (s *EndpointSuite) TestEndpointStatus(c *C) {
...
+	// Test that inserting identity labels works
+	rev := e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "bar", "zip": "zop"}, "cilium"))
+	c.Assert(rev, Not(Equals), 0)
+	c.Assert(string(e.OpLabels.OrchestrationIdentity.SortedList()), Equals, "cilium:foo=bar;cilium:zip=zop;")
+	// Test that nothing changes
+	rev = e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "bar", "zip": "zop"}, "cilium"))
+	c.Assert(rev, Equals, 0)
+	c.Assert(string(e.OpLabels.OrchestrationIdentity.SortedList()), Equals, "cilium:foo=bar;cilium:zip=zop;")
+	// Remove one label, change the source and value of the other.
+	rev = e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "zop"}, "ngnix"))
```
Comment (joestringer) [inline pkg/endpoint/endpoint_test.go]: `nginx` is misspelled ;-)

**Review by joestringer — APPROVED** [review]
Looks good to me.

File: pkg/endpoint/endpoint_test.go (hunk 1 above)
Comment (borkmann) [reply pkg/endpoint/endpoint_test.go]: @jrajahalme Could you still fix this before merge? Thx

Comment (jrajahalme) [reply pkg/endpoint/endpoint_test.go]: Fixed.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Pull request merged**